packages/accounts-password/password_client.js

```diff
@@ -25,12 +25,15 @@ const reportError = (error, callback) => {
  *   single key: `email`, `username` or `id`. Username or email match in a case
  *   insensitive manner.
  * @param {String} password The user's password.
+ * @param {String} token
+ *   Optional if not using the package accounts-2fa. This will be the user's token
+ *   when they're trying to log in.
  * @param {Function} [callback] Optional callback.
  *   Called with no arguments on success, or with a single `Error` argument
  *   on failure.
  * @importFromPackage meteor
  */
-Meteor.loginWithPassword = (selector, password, callback) => {
+Meteor.loginWithPassword = (selector, password, token, callback) => {
```

Comment:
This change is going to break all the codes that we have using Meteor.loginWithPassword, right?

This can't happen.

You need to check if the third arg is a function or not.

But I'm still not 100% sure this is the right approach.

Why didn't you create a new Meteor.loginWithPasswordAnd2fa and internally you call this loginWithPassword as well with different parameters but keeping the existent one untouched in terms of parameters?

It would have a new `internalLoginWithPassword` that could be even more flexible, receiving args as objects with more options and better long-term compatibility.

Reply:
I think coupling `accounts-password`and `accounts-2fa` together like this is not a good solution. It breaks all current apps using `accounts-password.` But it also ignores the flexibility that is already built into the accounts system.

Instead of modifying the existing password login handler, a better approach is to make use of Accounts.registerLoginHandler to register a new login handler on server side just for this package and then call that from client side with Accounts.callLoginMethod (or perhaps even just with Meteor.loginWithPassword). As I understand it, login handlers are executed one after another and can be stacked like this.

This approach is described in the following Meteor blog post: https://blog.meteor.com/tutorial-two-factor-authentication-with-meteor-and-totp-21d4a2f9ee51

Reply:
Logic changed as Filipe suggested in a 101 call ;)